cf-agent/simulate_mode.c

```diff
@@ -697,6 +697,64 @@ static void PkgOperationRecordDestroy(PkgOperationRecord *pkg_op)
     }
 }
 
+static inline bool PkgVersionIsGreater(const char *ver1, const char *ver2)
+{
+    /* Empty/missing versions should be handled separately based on the
+     * operations they appear in */
+    assert(!NULL_OR_EMPTY(ver1) && !NULL_OR_EMPTY(ver2));
+
+    /* "latest" is greater than any version */
+    if (StringEqual(ver1, "latest"))
+    {
+        return true;
+    }
+
+    /* TODO: do real version comparison */
+    return (StringSafeCompare(ver1, ver2) == 1);
```

Comment:
https://github.com/cfengine/libntech/commit/eff7196d26a841af40fbd8b36e1b6064bceb3e28

Could reuse / improve that. It expects 1-3 numbers separated by dots, and nothing else, currently.

Reply:
Yup.